### PR TITLE
http: removes the default-true use_new_codec_wrapper runtime feature.

### PR DESCRIPTION
**Pull request by birenroy**
Signed-off-by: Biren Roy <birenroy@google.com>

Commit Message: Resolves issue #23596.
Additional Description:
Risk Level: low
Testing: Ran unit tests and integration tests locally.
Docs Changes:
Release Notes: http: removes the `use_new_codec_wrapper` runtime feature
Platform Specific Features:

### CODE REVIEW

File: source/common/http/http2/codec_impl.cc [hunk 1]
```diff
@@ -1627,12 +1488,7 @@ Status ConnectionImpl::sendPendingFrames() {
     return okStatus();
   }
 
-  int rc;
-  if (use_new_codec_wrapper_) {
-    rc = adapter_->Send();
-  } else {
-    rc = nghttp2_session_send(session_);
-  }
+  int rc = adapter_->Send();
```
Comment (KBaichoo) [inline source/common/http/http2/codec_impl.cc:1491]: nit: const

**Review by KBaichoo — COMMENTED** [review]
Hey @birenroy,

Thanks for the clean up! Can you merge main to resolve the merge conflict?

Doing a grep for `http2_new_codec_wrapper` I think we're missing changes to test/common/http/codec_impl_fuzz_test.cc  and test/integration/h2_wrapped_capture_fuzz_test.cc.

File: source/common/http/http2/codec_impl.cc (hunk 1 above)
Comment (birenroy) [reply source/common/http/http2/codec_impl.cc]: Done.

File: source/common/http/http2/codec_impl.h [hunk 2]
```diff
@@ -602,14 +579,9 @@ class ConnectionImpl : public virtual Connection,
   void scheduleProtocolConstraintViolationCallback();
   void onProtocolConstraintViolation();
 
-  // Uses a new wrapper API around the underlying HTTP/2 codec. Guarded by the
-  // "envoy.reloadable_features.http2_new_codec_wrapper" runtime feature flag.
-  const bool use_new_codec_wrapper_;
   // Whether to use the new HTTP/2 library. Only has an effect if `use_new_codec_wrapper` is true.
```
[the remark this answers is not in the document]
Comment (birenroy) [reply source/common/http/http2/codec_impl.h]: Done.

File: test/integration/fake_upstream.cc [hunk 3]
```diff
@@ -333,19 +333,10 @@ class TestHttp2ServerConnectionImpl : public Http::Http2::ServerConnectionImpl {
...
-      rc = adapter_->Send();
-    } else {
-      absl::InlinedVector<nghttp2_settings_entry, 1> settings;
-      settings.insert(settings.end(), {{NGHTTP2_SETTINGS_MAX_CONCURRENT_STREAMS, max_streams}});
-      rc = nghttp2_submit_settings(session_, NGHTTP2_FLAG_NONE, settings.data(), settings.size());
-      ASSERT(rc == 0);
-      rc = nghttp2_session_send(session_);
-    }
+    absl::InlinedVector<http2::adapter::Http2Setting, 1> settings;
+    settings.insert(settings.end(), {{http2::adapter::MAX_CONCURRENT_STREAMS, max_streams}});
```
[the remark this answers is not in the document]
Comment (birenroy) [reply test/integration/fake_upstream.cc]: Done. (I'm pretty sure this predated my changes to the file.)

File: test/integration/fake_upstream.cc [hunk 4]
```diff
@@ -333,19 +333,10 @@ class TestHttp2ServerConnectionImpl : public Http::Http2::ServerConnectionImpl {
...
+    adapter_->SubmitSettings(settings);
+    int rc = adapter_->Send();
```
[the remark this answers is not in the document]
Comment (birenroy) [reply test/integration/fake_upstream.cc]: Done.

**Review by KBaichoo — APPROVED** [review]
Thanks for all the work on the HTTP/2 codec. lgtm

**Pull request merged**